api/src/database/run-ast.ts

```diff
@@ -244,12 +246,127 @@ function getDBQuery(
 	query: Query
 ): Knex.QueryBuilder {
 	const preProcess = getColumnPreprocessor(knex, schema, table);
-	const dbQuery = knex.select(fieldNodes.map(preProcess)).from(table);
 	const queryCopy = clone(query);
 
 	queryCopy.limit = typeof queryCopy.limit === 'number' ? queryCopy.limit : 100;
 
-	return applyQuery(knex, table, dbQuery, queryCopy, schema);
+	// Queries with aggregates and groupBy will not have duplicate result
+	if (queryCopy.aggregate || queryCopy.group) {
+		const flatQuery = knex.select(fieldNodes.map(preProcess)).from(table);
+		return applyQuery(knex, table, flatQuery, queryCopy, schema).query;
+	}
+
+	const primaryKey = schema.collections[table].primary;
+	const aliasMap: AliasMap = Object.create(null);
+	const dbQuery = knex.from(table);
+	let sortRecords: ColumnSortRecord[] | undefined;
+	const innerQuerySortRecords: { alias: string; order: 'asc' | 'desc' }[] = [];
+	let hasMultiRelationalSort: boolean | undefined;
+
+	if (queryCopy.sort) {
+		({ sortRecords, hasMultiRelationalSort } =
+			applySort(knex, schema, dbQuery, queryCopy.sort, table, aliasMap, true) ?? {});
```

Comment:
```suggestion
                const sortResult = applySort(knex, schema, dbQuery, queryCopy.sort, table, aliasMap, true);
                if (sortResult) {
                    sortRecords = sortResult.sortRecords;
                    hasMultiRelationalSort = sortResult.hasMultiRelationalSort;
                }
```
This one also feels a bit dense for a oneliner